pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -205,6 +205,7 @@
     private boolean autoReadDisabledPublishBufferLimiting = false;
     private final long maxPendingBytesPerThread;
     private final long resumeThresholdPendingBytesPerThread;
+    public static boolean autoShrinkForConsumerPendingAcksMap;
```

Comment:
this shouldn't be static

Reply:
Do you have any good ideas？

Reply:
I define static variables in Consumer and initialize them in ServerCnx, what do you think? like this:
![image](https://user-images.githubusercontent.com/19296967/156386314-99cd4f59-f9f6-4c13-8133-fabe1a6518ec.png)
![image](https://user-images.githubusercontent.com/19296967/156386362-62e0a1dc-d9d9-47df-a441-ac69255a37bf.png)

Reply:
```suggestion
    @Getter
    private boolean autoShrinkForConsumerPendingAcksMap;
```
I think this works.

Reply:
The problem here is: if you don't define autoShrinkForConsumerPendingAcksMap as static, you can only pass it to the Consumer class by way of method parameters.

Reply:
> I think this works.

The question here is: how do we pass autoShrinkForConsumerPendingAcksMap to the Consumer class, if we don't define autoShrinkForConsumerPendingAcksMap as static and pass the method parameters.

Reply:
Aha, I found a solution, the configuration can be accessed in this way in the consumer class:
subscription.getTopic().getBrokerService()
                           .getPulsar().getConfiguration().isAutoShrinkForConsumerPendingAcksMap()

Reply:
Fixed.
@eolivelli @Jason918 PTAL,thanks